src/Storage/AuthorizationCodeStorage.php

```diff
@@ -5,120 +5,88 @@
 use OAuth2\OpenID\Storage\AuthorizationCodeInterface;
 
 class AuthorizationCodeStorage implements AuthorizationCodeInterface {
-	const TAXONOMY = 'oidc-authorization-code';
+	const META_KEY_PREFIX = 'oidc';
```

Comment:
I am not sure I follow how it would have the same problem here without this data being namespaced by client. The retrieval happens here by auth code itself, which would only be in possession of the intended oidc client. Can you explain?

> The [meta key length of 255](https://github.com/WordPress/WordPress/blob/master/wp-admin/includes/schema.php#L232) should allow us to do that and at the same time eliminate the slow query problem.

Sorry, don't understand what you mean here. Can you elaborate?

Reply:
> I am not sure I follow how it would have the same problem here without this data being namespaced by client. The retrieval happens here by auth code itself, which would only be in possession of the intended oidc client. Can you explain?

Try this mental experiement: what if two OpenID clients request authorization in short consequence. The authcode of client 1 would be overwritten by client 2.

> Sorry, don't understand what you mean here. Can you elaborate?

Please take a look at [my comment here](https://github.com/Automattic/wp-openid-connect-server/pull/45#discussion_r988872553). I think you have just been ignoring the phpcs error without trying to understand the message. Querying by `meta_value` is slow, especially if you anticipate running this on a server with thousands of users.

Reply:
> Try this mental experiement: what if two OpenID clients request authorization in short consequence. The authcode of client 1 would be overwritten by client 2.

Yes! But is this an actual need where a user would start 2nd OIDC client authentication before finishing with 1st OIDC client authentication? If you would like to support that, I would understand this needs to change. As of now, it would just fail to login both clients and only the latter one would end up winning (successful login).

> Please take a look at https://github.com/Automattic/wp-openid-connect-server/pull/45#discussion_r988872553.

I responded there after responding to this comment.

> I think you have just been ignoring the phpcs error without trying to understand the message.

Negative!

I didn't understand your comment about meta key length & namespaced data eliminating the slow query problem.

Reply:
> Yes! But is this an actual need where a user would start 2nd OIDC client authentication before finishing with 1st OIDC client authentication? If you would like to support that, I would understand this needs to change. As of now, it would just fail to login both clients and only the latter one would end up winning (successful login).

If there is a race condition then the details of two oidc clients could end up being stored mixed. I see it as an unclean implementation when we store related values in a way where we don't ensure that they are stored or deleted together.

Reply:
> If our code is less than 255 minus key namespace (e.g. oidc_code_<code_in_verbatim> would give us 245 bytes for the code), then we can look up a code just using a meta_key query which is not slow because it is indexed.

That makes sense. Thanks for explaining!